custom_components/tuya_local/devices/dr_heater_dr008_thermostat.yaml

```diff
@@ -0,0 +1,229 @@
+  - entity: lock
+    name: Screen lock
+    category: config
+    dps:
+      - id: 108
+        type: boolean
+        name: lock
+        mapping:
+          - dps_val: true
+            icon: "mdi:hand-back-right-off"
+          - dps_val: false
+            icon: "mdi:hand-back-right"
+  - entity: sensor
+    name: Room temperature
+    class: temperature
+    icon: "mdi:home-thermometer"
+    dps:
+      - id: 105
+        type: integer
+        name: sensor
+        class: measurement
+        mapping:
+          - scale: 10
+      - id: 107
+        name: unit
+        type: boolean
+        mapping:
+          - dps_val: false
+            value: C
+          - dps_val: true
+            value: F
+  - entity: sensor
+    name: Floor temperature
+    class: temperature
+    icon: "mdi:thermometer-lines"
+    dps:
+      - id: 106
+        type: integer
+        name: sensor
+        class: measurement
+        mapping:
+          - scale: 10
+      - id: 107
+        name: unit
+        type: boolean
+        mapping:
+          - dps_val: false
+            value: C
+          - dps_val: true
+            value: F
+  - entity: sensor
+    name: Energy
+    dps:
+      - id: 117
+        type: integer
+        name: sensor
+        unit: kWh
+        mapping:
+          - scale: 10
+      - id: 116
+        type: integer
+        name: calibration
+  - entity: select
+    category: config
+    translation_key: temperature_unit
+    dps:
+      - id: 107
+        name: option
+        type: boolean
+        mapping:
+          - dps_val: false
+            value: celsius
+          - dps_val: true
+            value: fahrenheit
+  - entity: number
+    category: config
+    class: temperature
+    name: Room temperature limit
+    icon: "mdi:thermometer-chevron-down"
+    dps:
+      - id: 121
+        type: integer
+        name: value
+        range:
+          min: 200
+          max: 500
+        mapping:
+          - scale: 10
+            step: 5
+            constraint: unit
+            conditions:
+              - dps_val: true
+                step: 30
+                range:
+                  min: 680
+                  max: 1220
+      - id: 107
+        name: unit
+        type: boolean
+        mapping:
+          - dps_val: false
+            value: C
+          - dps_val: true
+            value: F
+  - entity: select
+    category: config
+    name: Temperature sensor
+    icon: "mdi:thermometer-check"
+    dps:
+      - id: 111
+        type: string
+        name: option
+        mapping:
+          - dps_val: "0"
+            value: "Room"
+            icon: "mdi:home-thermometer"
+          - dps_val: "1"
+            value: "Floor"
+            icon: "mdi:heating-coil"
+          - dps_val: "2"
+            value: "Both"
+            icon: "mdi:thermometer"
+  - entity: select
+    category: config
+    name: Auto schedule
+    icon: "mdi:calendar-sync"
+    dps:
+      - id: 110
+        type: integer
```

Comment:
Is it really an integer, and not a string?

Reply:
Good question. As I mentioned above, this started as a clone of [Jiahong ET-72W thermostat](https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/jiahong_et72w_thermostat.yaml), and I only changed what I was familiar with. Everything else seemed alright, so I didn't mess with it.

Reply:
Usually these "select between a limited list of options" are what Tuya calls "enum" type, which is a string. The string may be numeric values, but has the quotes around it.

Because this has been confusing in the past, the current code is designed to work if you make them integer, but at the expense of some false matches during device detection, as the config will be matching both integers and quoted integers, while if you list them as string it will only match the quoted ones. Also it isn't guaranteed that this leniency will stay around forever, if we ever find out it causes a real problem somewhere.

Reply:
Makes sense. I'll try and reconcile it with actual values.